frontend/src/filter_table.tsx

```diff
@@ -165,25 +174,150 @@ export function FilterTable({individuals, title = '', filters = [],
           return false;
         }
       }
+
+      if (search) {
+        // simplest possible search for now - filter everything that doesn't
+        // have an active field that matches the search term
+        let searchResult = false
+        const searchRegExp = new RegExp(search, 'i')
+        for (let column of visibleColumns) {
+          if (!i[column.field]) {
+            continue
+          }
+          const value = column.sortBy ? `${i[column.field][column.sortBy]}`
+                                      : `${i[column.field]}`
+          if (value.match(searchRegExp)) {
+            searchResult = true
+            break
+          }
+        }
+        return searchResult
+      }
+
       return true
     })
-  }
+  }, [filters, individuals, search])
+
+  /**
+   * Memoized list of individuals sorted by `orderBy` and `order`, as well as
+   * the column definition information `sortBy` and `sortAs`. `orderBy` must be
+   * a field in `allColumnns`.
+   * When the value of a field is an object, `sortBy` is used to select which
+   * subfield to use for sorting. `sortAs` can be `number`, `numbers` or `date`
+   * to force a column to be sorted as a particular type.
+   * - `date` fields will be cast as dates using `new Date(<value)`
+   * - `number` fields will be cast as numbers by removing all non-numerical
+   *   characters. ex. `G111-222` -> `111222`
+   * - `numbers` fields are cast as numbers, but using the first set of numbers
+   *   as the integer part, and the remaning as decimals.
+   *   ex. `G111-222-333` -> `111.222333`.
+   */
+  const sortedIndividuals = React.useMemo(() => {
```

Comment:
I would go for putting the sort function on the top level or in a separate module so it's easier to test.